From b9ec23b25cd934234594e06a11eeff2ed12ab052 Mon Sep 17 00:00:00 2001
From: Luciano Coelho <coelho@ti.com>
Date: Fri, 30 Sep 2011 16:20:03 +0300
Subject: [PATCH 08/29] nl80211: check if sched_scan is supported in wiphy info

Tell wpa_supplicant whether sched_scan is supported by the driver by
looking into the wiphy info data.

Signed-off-by: Luciano Coelho <coelho@ti.com>
---
 src/drivers/driver_nl80211.c |    5 +++++
 1 files changed, 5 insertions(+), 0 deletions(-)

diff --git a/src/drivers/driver_nl80211.c b/src/drivers/driver_nl80211.c
index 3025b49..bc11fb9 100644
--- a/src/drivers/driver_nl80211.c
+++ b/src/drivers/driver_nl80211.c
@@ -1570,6 +1570,7 @@ struct wiphy_info_data {
 	int connect_supported;
 	int offchan_tx_supported;
 	int max_remain_on_chan;
+	int sched_scan_supported;
 };
 
 
@@ -1623,6 +1624,8 @@ static int wiphy_info_handler(struct nl_msg *msg, void *arg)
 				info->auth_supported = 1;
 			else if (cmd == NL80211_CMD_CONNECT)
 				info->connect_supported = 1;
+			else if (cmd == NL80211_CMD_START_SCHED_SCAN)
+				info->sched_scan_supported = 1;
 		}
 	}
 
@@ -1686,6 +1689,8 @@ static int wpa_driver_nl80211_capa(struct wpa_driver_nl80211_data *drv)
 
 	drv->capa.max_scan_ssids = info.max_scan_ssids;
 	drv->capa.max_sched_scan_ssids = info.max_sched_scan_ssids;
+	drv->capa.sched_scan_supported = info.sched_scan_supported;
+
 	if (info.ap_supported)
 		drv->capa.flags |= WPA_DRIVER_FLAGS_AP;
 
-- 
1.7.0.4

